src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -150,54 +150,198 @@ begin
   simp [e₂, direct_sum.submodule_coe, direct_sum.to_module, dfinsupp.sum_add_hom_apply]
 end
 
-/-- An orthonormal basis on a fintype `ι` for an inner product space induces an isometry with
+variables (ι) (𝕜) (E)
+
+/-- The vector given in euclidean space by being `1 : 𝕜` at coordinate `i : ι` and `0 : 𝕜` at
+all other coordinates. -/
+def euclidean_space.single {𝕜 : Type*} {ι : Type*} [fintype ι] [is_R_or_C 𝕜] (i : ι) (a : 𝕜) :
+ euclidean_space 𝕜 ι :=
+  set.indicator {i} (λ j, a)
```

Comment:
Likely better as
```suggestion
pi.single i a
```

Reply:
Agreed, sorry to make you change it twice, but when I suggested `set.indicator` before I'd forgotten we had the more on-the-nose `pi.single`.